Source/Urho3D/Container/Str.h

```diff
@@ -155,9 +164,20 @@ class URHO3D_API String
     /// Assign a string.
     String& operator =(const String& rhs)
     {
-        Resize(rhs.length_);
-        CopyChars(buffer_, rhs.buffer_, rhs.length_);
+        if (&rhs != this)

```

Comment:
Maybe this should be `assert`? That would be consistent with other operators. Besides there is no reason for user to ever do this anyway.